src/WebApp/appsettings.json

```diff
@@ -0,0 +1,25 @@
+{
+    "Logging": {
+        "Default": "Debug",
+        "Microsoft": "Warning",
+        "Microsoft.Hosting.Lifetime": "Information"
+    },
+    "IdentityServer": {
+        "Clients": [
+            {
+                "ClientId": "codidact.com",
+                "ClientSecrets": [
+                    {
+                        "Value": "LCa0a2j/xo/5m0U8HTBBNBNCLXBkg7+g+YpeiGJm564="
```

Comment:
Put a placeholder here

Reply:
That's the SHA256 for 'foo'. It looks like that because it's encoded in base64.

Reply:
Well technically, security wise its not ok to keep this here. (I am not talking about foo, I am talking about the key/value)
I am not asking you to change this now but take a look at what I did on https://github.com/codidact/core/pull/60

https://github.com/codidact/core/blob/5b2b6c273e60e7e8b75b96d01aa56ade1c5642de/src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs
https://github.com/codidact/core/blob/5b2b6c273e60e7e8b75b96d01aa56ade1c5642de/src/WebUI/Startup.cs

Reply:
86d35004bb522a17518d42566a3923884dd2c715

Reply:
@misha130 I got a different idea, we could simply create an `ISecretConfiguration` interface that is essentially an alias for `IConfiguration` and have it resolve a different configuration than the normal `IConfiguration`. In a development environment we could simply make it an alias for the normal `IConfiguration`:

f842cab2978f035f68fec0da0632b71be238b09b

The problem is that this removes the possibility for using asynchronous methods because the `IConfiguration` interface does not supply them. Do you expect that loading the secret configuration is an expensive operation that needs `async`/`await`?

Reply:
ISecretService is not an alias for IConfiguration.
Only secrets should be taken from ISecretService.
This is why its async because it might be remote.

The whole issue came up when I tried storing the connection string in a dev secrets rep realizing I can't use it in production. So I created this to abstract what is used. In the future the secrets should be taken from the cloud or DB.

This is how I originally planned it at least.

Reply:
https://ptb.discordapp.com/channels/634104110131445811/635510829499940912/684078492773777514